regions/_utils/optional_deps.py

```diff
@@ -3,11 +3,12 @@
 try:
     import matplotlib
     HAS_MATPLOTLIB = True
-    MPL_VERSION = getattr(matplotlib, '__version__', None)
-    if MPL_VERSION is None:
-        MPL_VERSION = matplotlib._version.version
-    MPL_VERSION = MPL_VERSION.split('.')
+    MPL_VER_STR = getattr(matplotlib, '__version__', None)
+    if MPL_VER_STR is None:
+        MPL_VER_STR = matplotlib._version.version
+    MPL_VERSION = MPL_VER_STR.split('.')
     MPL_VERSION = 10 * int(MPL_VERSION[0]) + int(MPL_VERSION[1])
 except ImportError:
     HAS_MATPLOTLIB = False
+    MPL_VER_STR = 'None'
```

Comment:
Is is possible to streamline the matplotilb version checking?   Something like what astropy uses for numpy version checks (see https://github.com/astropy/astropy/blob/main/astropy/utils/compat/numpycompat.py)?
```
from astropy.utils import minversion

MPL_LT_3_6 = not minversion(matplotlib, '3.6')
```

and then use it the code with:
```
if MPL_LT_3_6:
  ...
```

Reply:
Thanks for reviewing; will look into it.
Want to check first if this can be made to work with matplotlib/matplotlib#21945, which I've just reopened as matplotlib/matplotlib#26833 – unfortunately switching to display coordinates seems to break not just rotation but our existing resize and move operations; getting errors about `self.height = ymax - ymin` in `_update_from_mpl_selector` being set to negative values...

Reply:
Thanks.  Let me know when this is ready for a (final?) review.